source/postprocess/interface.cc

```diff
@@ -268,6 +268,101 @@ namespace aspect
                 postprocessor_names.push_back (*p);
             }
         }
+
+      // we now have matching lists 'postprocessors' and 'postprocessor_names'
+      // that define which postprocessors we have. we just need to bring them
+      // into an order so that dependencies run first, and dependents after
+      // them. the algorithm for creating a sorted list for this works as follows:
+      //
+      // while there are postprocessors not yet added to the sorted list:
+      // - go through the list
+      // - if we encounter a postprocessor that has not yet been added yet:
+      //   . if all of its dependencies are in the list, add it to the end
+      //   . if at least one of its dependencies are not yet in the list,
+      //     skip it
+      //
+      // if we go through a loop where we do not add a postprocessor to the list
+      // but there are still ones that haven't been added to the list, then we
+      // have found a cycle in the dependencies and that is clearly a problem
+      std::vector<bool> already_assigned (postprocessors.size(), false);
+      std::vector<std::string> sorted_names;
+      while (sorted_names.size() < postprocessors.size())
+        {
+          bool at_least_one_element_added = false;
+
+          {
+            typename std::vector<std_cxx11::shared_ptr<Interface<dim> > >::const_iterator
+            pp = postprocessors.begin();
+            for (unsigned int i=0; i<postprocessor_names.size(); ++i, ++pp)
```

Comment:
Yes, they are constructed in lockstep just a couple dozen lines above. I'll add the assertion.